manager/apis/app/v1alpha1/blueprint_types.go

```diff
@@ -31,6 +31,11 @@ type ReadModuleArgs struct {
 	// +required
 	Source DataStore `json:"source"`
 
+	// AssetID identifies the asset to be used for accessing the data when it is ready
+	// It is copied from the M4DApplication resource
+	// +required
+	AssetID string `json:"assetID"`
```

Comment:
Shouldn't this be part of the DataStore struct?

Reply:
This is relevant only for the read module that includes this information in data access code. 
Conceptually, DataStore represents the "physical" data endpoint. If read module had a Destination as well as a Source, AssetID  would have been used inside Destination.

Reply:
In the (future) ingest scenario we will have situations where we need to provide the asset ID of both the source and the destination, so I would suggest we put it in the DataStore - at least as an optional field.  Lineage will need this information as well

Reply:
@simanadler I'm not sure if this is the right way to specify this for the ingest flow target. Would it be okay to keep this PR as is and revisit after finalizing the ingest flow?

Reply:
@simanadler If I understand you correctly, you suggest to connect the connection details with a cataloged asset. If a copy module catalogs the implicit copy, and a read is applied after a copy, it will actually read a newly cataloged asset. However, from the perspective of a person who writes the code that interacts with the read module, it will be more work to copy a new access instructions each time a cataloged version changes. 
I agree that we'll need a new field to connect the data store with the cataloged version but not for the purpose of this PR.

Reply:
OK